backend/src/authorisation.js

```diff
@@ -0,0 +1,27 @@
+import { allow, rule, shield } from 'graphql-shield'
+
+const isAuthenticated = rule({ cache: 'contextual' })(
+  async (_, __, {
+    userId,
+    dataSources
+  }) => {
+    return userId !== null && dataSources.db.hasUser(userId)
+  }
+)
+
+const permissions = shield({
+  Query: {
+    posts: allow,
+    users: isAuthenticated
+  },
+  Mutation: {
+    createPost: isAuthenticated,
+    deletePost: isAuthenticated,
+    upvotePost: isAuthenticated,
+    downvotePost: isAuthenticated,
```

Comment:
You could use a fallbackRule for this so any Mutation or Query you add will use this rule by default.